metadata/src/main/java/com/redhat/lightblue/metadata/CompositeMetadata.java

```diff
@@ -445,6 +445,8 @@ private static void copyFields(Fields dest,
                         path.pop();
                     } else {
                         newElement = new SimpleArrayElement(((SimpleArrayElement) sourceEl).getType());
+                        ((SimpleArrayElement)newElement).setConstraints( ((SimpleArrayElement)sourceEl).getConstraints());
```

Comment:
You could avoid the cast by new'ing SimpleArrayElement as it's own type, then setting it as newElement on the way out.